tests/integration/hugectr/test_nvt_hugectr_training.py

```diff
@@ -0,0 +1,380 @@
+    slot_sizes = []
+    for column in CATEGORICAL_COLUMNS:
+        slot_sizes.append(embeddings[column][0])
+        total_cardinality += embeddings[column][0]
+
+    test_data_path = DATA_DIR + "test/"
+    if path.exists(test_data_path):
+        shutil.rmtree(test_data_path)
+
+    os.mkdir(test_data_path)
+
+    sample_data = cudf.read_parquet(DATA_DIR + "valid.parquet", num_rows=test_nrows)
+    sample_data.to_csv(test_data_path + "data.csv")
+
+    sample_data_trans = nvt.workflow._transform_partition(sample_data, [workflow.column_group])
+
+    dense_features, embedding_columns, row_ptrs = _convert(sample_data_trans, slot_sizes)
+
+    _run_model(slot_sizes, total_cardinality)
+
+    if path.exists(TEMP_DIR):
+        shutil.rmtree(TEMP_DIR)
+
+    os.mkdir(TEMP_DIR)
+
+    file_names = glob.iglob(os.path.join(os.getcwd(), "*.model"))
+    for files in file_names:
+        shutil.move(files, TEMP_DIR)
+
+    _write_model_json(slot_sizes, total_cardinality)
+
+    if path.exists(MODEL_DIR):
+        shutil.rmtree(MODEL_DIR)
+
+    os.mkdir(MODEL_DIR)
+
+    model_name = "test_model"
+    hugectr_params = dict()
+    hugectr_params["config"] = MODEL_DIR + "test_model/1/model.json"
+    hugectr_params["slots"] = len(slot_sizes)
+    hugectr_params["max_nnz"] = len(slot_sizes)
+    hugectr_params["embedding_vector_size"] = 16
+    hugectr_params["n_outputs"] = 1
+
+    export_hugectr_ensemble(
+        workflow=workflow,
+        hugectr_model_path=TEMP_DIR,
+        hugectr_params=hugectr_params,
+        name=model_name,
+        output_path=MODEL_DIR,
+        label_columns=["rating"],
+        cats=CATEGORICAL_COLUMNS,
+        max_batch_size=64,
+    )
+
+    shutil.rmtree(TEMP_DIR)
+
+    _predict(dense_features, embedding_columns, row_ptrs, hugectr_params["config"], model_name)
+
+
+def _run_model(slot_sizes, total_cardinality):
+
+    solver = hugectr.solver_parser_helper(
+        vvgpu=[[0]],
+        max_iter=2000,
+        batchsize=2048,
+        display=100,
+        eval_interval=200,
+        batchsize_eval=2048,
+        max_eval_batches=160,
+        i64_input_key=True,
+        use_mixed_precision=False,
+        repeat_dataset=True,
+        snapshot=1900,
+    )
+
+    optimizer = hugectr.optimizer.CreateOptimizer(
+        optimizer_type=hugectr.Optimizer_t.Adam, use_mixed_precision=False
+    )
+    model = hugectr.Model(solver, optimizer)
+
+    model.add(
+        hugectr.Input(
+            data_reader_type=hugectr.DataReaderType_t.Parquet,
+            source=DATA_DIR + "train/_file_list.txt",
+            eval_source=DATA_DIR + "valid/_file_list.txt",
+            check_type=hugectr.Check_t.Non,
+            label_dim=1,
+            label_name="label",
+            dense_dim=0,
+            dense_name="dense",
+            slot_size_array=slot_sizes,
+            data_reader_sparse_param_array=[
+                hugectr.DataReaderSparseParam(hugectr.DataReaderSparse_t.Distributed, 3, 1, 2)
```

Comment:
May be this line is the issue: (3, 1, 2)

would it work if you change it to `(3, 1, 3)` or `(4, 1, 3)` ?

Reply:
(4, 1, 3) worked. Thanks!